src/Plutarch/Test/QuickCheck/Modifiers.hs

```diff
@@ -292,6 +307,160 @@ instance Function (GenValue adaMod otherMod) where
   {-# INLINEABLE function #-}
   function = functionMap unwrap rewrap
 
+{- | Represents a change in 'POSIXTime'. The @mod@ argument gives the kind of
+ change, represented by a QuickCheck modifier, while the @n@ argument is a
+ closed (inclusive) upper bound on the magnitude of the change.
+
+ For example, @'TimeDelta' 'Positive' 100@ represents a change in 'POSIXTime'
+ from @1@ to @100@ units, while @'TimeDelta' 'NonPositive' 250@ represents a
+ change in 'POSIXTime' from @0@ to @-250@. Modifiers intended to work with
+ this type are:
+
+ - 'Positive'
+ - 'Negative'
+ - 'NonPositive'
+ - 'NonNegative'
+
+ The instances for 'TimeDelta' reflect this decision: while other modifiers
+ could potentially be useful, this ensures that we have safe behaviour and
+ efficient instances.
+
+ Shrinking a 'TimeDelta' will shrink towards zero: more specifically, it will
+ reduce the magnitude of the change.
+
+ To control what 'TimeDelta' you get, the easiest method is a type signature:
+
+ > forAll arbitrary $ \(delta :: TimeDelta NonNegative 100) -> ...
+
+ @since 2.1.4
+-}
+newtype TimeDelta (mod :: Type -> Type) (n :: Nat)
+  = TimeDelta (mod POSIXTime)
+
+-- | @since 2.1.4
+deriving via
+  (mod POSIXTime)
+  instance
+    (forall (a :: Type). (Eq a) => Eq (mod a)) =>
+    Eq (TimeDelta mod n)
+
+-- | @since 2.1.4
+deriving stock instance
+  (forall (a :: Type). (Show a) => Show (mod a)) =>
+  Show (TimeDelta mod n)
+
+{- | Strictly positive deltas are semigroups under addition.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Semigroup (TimeDelta Positive n)
+
+{- | Strictly negative deltas are semigroups under addition.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Semigroup (TimeDelta Negative n)
+
+{- | Non-negative deltas are semigroups under addition.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Semigroup (TimeDelta NonNegative n)
+
+{- | Non-positive deltas are semigroups under addition.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Semigroup (TimeDelta NonPositive n)
+
+{- | Non-negative deltas are monoids with the zero delta as the identity.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Monoid (TimeDelta NonNegative n)
+
+{- | Non-positive deltas are monoids with the zero delta as the identity.
+
+ @since 2.1.4
+-}
+deriving via (Sum Integer) instance Monoid (TimeDelta NonPositive n)
+
+-- | @since 2.1.4
+instance (KnownNat n, 1 <= n) => Arbitrary (TimeDelta Positive n) where
```

Comment:
Perhaps, an instance for `Arbitrary (TimeDelta Identity n)` where it generates from entire domain?

Reply:
I thought about that, as well as `NonZero` for the entire domain excluding zero. Will add.